mesonbuild/interpreter.py

```diff
@@ -512,10 +512,16 @@ def partial_dependency_method(self, args, kwargs):
         return DependencyHolder(pdep, self.subproject)
 
     @FeatureNew('dep.get_variable', '0.51.0')
-    @noPosargs
+    @typed_pos_args('dep.get_variable', optargs=[str])
     @permittedKwargs({'cmake', 'pkgconfig', 'configtool', 'internal', 'default_value', 'pkgconfig_define'})
     @FeatureNewKwargs('dep.get_variable', '0.54.0', ['internal'])
-    def variable_method(self, args, kwargs):
+    def variable_method(self, args: T.Tuple[T.Optional[str]], kwargs: T.Dict[str, T.Any]) -> str:
+        default_varname = args[0]
+        if default_varname is not None:
+            FeatureNew('0.58.0', 'Positional argument to dep.get_variable()').use(self.subproject)
+            for k in ['cmake', 'pkgconfig', 'configtool', 'internal']:
+                if k not in kwargs:
+                    kwargs[k] = default_varname
```

Comment:
you could simplify this to
I think you can simplify this to:
```suggestion
            for k in ['cmake', 'pkgconfig', 'configtool', 'internal']:
                kwargs.setdefault(k, default_varname)
```